### ISSUE TITLE
Converting from terraform error

### ISSUE THREAD

**Issue by roderik** [pulumi/pulumi#13191]
### What happened?

Tried to convert a terraform project

```
Pulumi Version:   v3.72.1
error: vpc.pp:0,0-48,2: /var/folders/wj/n2px6dr928v4fzz1hhgbz3g80000gn/T/pulumi-convert1619114849/vpc_3.14.4/main.pp:168,5-69: cannot iterate over a value of type number; , and 3 other diagnostic(s); /var/folders/wj/n2px6dr928v4fzz1hhgbz3g80000gn/T/pulumi-convert1619114849/vpc_3.14.4/main.pp:168,5-69: cannot iterate over a value of type number; , and 3 other diagnostic(s)
error: rds.pp:4,1-34,2: /var/folders/wj/n2px6dr928v4fzz1hhgbz3g80000gn/T/pulumi-convert1619114849/rds-aurora_7.1.0/main.pp:111,5-62: cannot iterate over a value of type union(dynamic, object({})); ; /var/folders/wj/n2px6dr928v4fzz1hhgbz3g80000gn/T/pulumi-convert1619114849/rds-aurora_7.1.0/main.pp:111,5-62: cannot iterate over a value of type union(dynamic, object({}));
error: rds.pp:47,1-81,2: /var/folders/wj/n2px6dr928v4fzz1hhgbz3g80000gn/T/pulumi-convert1619114849/rds-aurora_7.1.0/main.pp:111,5-62: cannot iterate over a value of type union(dynamic, object({})); ; /var/folders/wj/n2px6dr928v4fzz1hhgbz3g80000gn/T/pulumi-convert1619114849/rds-aurora_7.1.0/main.pp:111,5-62: cannot iterate over a value of type union(dynamic, object({}));
error: rds.pp:94,1-129,2: /var/folders/wj/n2px6dr928v4fzz1hhgbz3g80000gn/T/pulumi-convert1619114849/rds-aurora_7.1.0/main.pp:111,5-62: cannot iterate over a value of type union(dynamic, object({})); ; /var/folders/wj/n2px6dr928v4fzz1hhgbz3g80000gn/T/pulumi-convert1619114849/rds-aurora_7.1.0/main.pp:111,5-62: cannot iterate over a value of type union(dynamic, object({}));
error: rds.pp:142,1-175,2: /var/folders/wj/n2px6dr928v4fzz1hhgbz3g80000gn/T/pulumi-convert1619114849/rds-aurora_7.1.0/main.pp:111,5-62: cannot iterate over a value of type union(dynamic, object({})); ; /var/folders/wj/n2px6dr928v4fzz1hhgbz3g80000gn/T/pulumi-convert1619114849/rds-aurora_7.1.0/main.pp:111,5-62: cannot iterate over a value of type union(dynamic, object({}));
error: rds.pp:191,1-223,2: /var/folders/wj/n2px6dr928v4fzz1hhgbz3g80000gn/T/pulumi-convert1619114849/rds-aurora_7.1.0/main.pp:111,5-62: cannot iterate over a value of type union(dynamic, object({})); ; /var/folders/wj/n2px6dr928v4fzz1hhgbz3g80000gn/T/pulumi-convert1619114849/rds-aurora_7.1.0/main.pp:111,5-62: cannot iterate over a value of type union(dynamic, object({}));
error: rds.pp:298,1-328,2: /var/folders/wj/n2px6dr928v4fzz1hhgbz3g80000gn/T/pulumi-convert1619114849/rds-aurora_7.1.0/main.pp:111,5-62: cannot iterate over a value of type union(dynamic, object({})); ; /var/folders/wj/n2px6dr928v4fzz1hhgbz3g80000gn/T/pulumi-convert1619114849/rds-aurora_7.1.0/main.pp:111,5-62: cannot iterate over a value of type union(dynamic, object({}));
error: could not generate output program
```

### Expected Behavior

It should work

### Steps to reproduce

I cannot share the terraform project, I'm afraid, but if you tell me what to look for I can probably produce some snippets

### Output of `pulumi about`

```
CLI
Version      3.72.1
Go Version   go1.20.5
Go Compiler  gc

Host
OS       darwin
Version  14.0
Arch     arm64

Backend
Name           pulumi.com
URL            https://app.pulumi.com/roderik
User           roderik
Organizations  roderik, settlemint

Pulumi locates its logs in /var/folders/wj/n2px6dr928v4fzz1hhgbz3g80000gn/T/ by default
warning: Failed to read project: no Pulumi.yaml project file found (searching upwards from /Users/roderik/Downloads/zkevm-terraform-main). If you have not created a project yet, use `pulumi new` to do so: no project file found
warning: Failed to get information about the current stack: no Pulumi.yaml project file found (searching upwards from /Users/roderik/Downloads/zkevm-terraform-main). If you have not created a project yet, use `pulumi new` to do so: no project file found
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by Zaid-Ajaj** [comment]
Hi there @roderik, thanks a lot for filing the issue! Your feedback is appreciated as we are trying to fix as many convert problems as possible. From the error it looks like you are trying to iterate over a value that has a bound type `union(dynamic, object({}))` which looks quite odd to be honest. 

Is there any chance you could a snippet of the code for us to debug? If possible the definition of the value you are iterating over and the location where it is being used.

**Comment by Zaid-Ajaj** [comment]
Hi @roderik this issue is fixed by #13459 and the conversion should work when the next version of Pulumi CLI comes out (probably v3.76.0) I'll keep the issue open until then

**Comment by Zaid-Ajaj** [comment]
Closing as fixed by #13459 which is now available as part Pulumi CLI v3.76.0 please consider re-opening if the problem persists
